edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/MetricsModule.cs

```diff
@@ -18,18 +18,20 @@ public sealed class MetricsModule : Module
         MetricsConfig metricsConfig;
         string iothubHostname;
         string deviceId;
+        string edgeAgentStorageFolder;
 
-        public MetricsModule(MetricsConfig metricsConfig, string iothubHostname, string deviceId)
+        public MetricsModule(MetricsConfig metricsConfig, string iothubHostname, string deviceId, string edgeAgentStorageFolder)
         {
             this.metricsConfig = Preconditions.CheckNotNull(metricsConfig, nameof(metricsConfig));
             this.iothubHostname = Preconditions.CheckNotNull(iothubHostname, nameof(iothubHostname));
             this.deviceId = Preconditions.CheckNotNull(deviceId, nameof(deviceId));
+            this.edgeAgentStorageFolder = Preconditions.CheckNotNull(edgeAgentStorageFolder, nameof(edgeAgentStorageFolder));
```

Comment:
what happens if this path doesn't exist?

Reply:
I believe this would cause a problem downstream when attempting to create the inner storage directory (not the same one) in MetricsProvider.